lib/ansible/modules/cloud/amazon/lambda_policy.py

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/python
+# Copyright (c) 2016, Pierre Jodouin <pjodouin@virtualcomputing.solutions>
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: lambda_policy
+short_description: Creates, updates or deletes AWS Lambda policy statements.
+description:
+    - This module allows the management of AWS Lambda policy statements.
+      It is idempotent and supports "Check" mode.  Use module M(lambda) to manage the lambda
+      function itself, M(lambda_alias) to manage function aliases, M(lambda_event) to manage event source mappings
+      such as Kinesis streams, M(lambda_invoke) to execute a lambda function and M(lambda_facts) to gather facts
+      relating to one or more lambda functions.
+
+version_added: "2.4"
+
+author:
+  - Pierre Jodouin (@pjodouin)
+  - Michael De La Rue (@mikedlr)
+options:
+  function_name:
+    description:
+      - "Name of the Lambda function whose resource policy you are updating by adding a new permission."
+      - "You can specify a function name (for example, Thumbnail ) or you can specify Amazon Resource Name (ARN) of the"
+      - "function (for example, arn:aws:lambda:us-west-2:account-id:function:ThumbNail ). AWS Lambda also allows you to"
+      - "specify partial ARN (for example, account-id:Thumbnail ). Note that the length constraint applies only to the"
+      - "ARN. If you specify only the function name, it is limited to 64 character in length."
+    required: true
+    aliases: ['lambda_function_arn', 'function_arn']
+
+  state:
+    description:
+      - Describes the desired state.
+    required: true
+    default: "present"
+    choices: ["present", "absent"]
+
+  alias:
+    description:
+      - Name of the function alias. Mutually exclusive with C(version).
+
+  version:
+    description:
+      -  Version of the Lambda function. Mutually exclusive with C(alias).
+    default: 0
+
+  statement_id:
+    description:
+      -  A unique statement identifier.
+    required: true
+    aliases: ['sid']
+
+  action:
```

Comment:
currently the module is limited to handling only one statement at a time; how would you propose to use a list?  Combine? Prepare for later?